### PR TITLE
Made a program to calculate options of right triangle

### PR DESCRIPTION
**Pull request by deniscvzz**

### CODE REVIEW

File: m2/main.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+print('Программа для нахождения гипотенузы, периметра, площади и углов прямоугольного треугольника.')
+print('Введите значения катетов в сантиметрах:')
+
+a = float(input("a = "))
+b = float(input("b = "))
+
+c, p, s, ga, al, be = right_triangle(a, b)
+
+if a == 0 or b == 0:
```
Comment (screwyprof) [inline m2/main.py]: To check if the given sides make a triangle the following formula can be used:
(a + b) > c && (a + c) > b && (b + c) > a

File: m2/main.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    print("a = 0 или b == 0, такого треугольника не существует.")
+    exit_with_msg(0)
+
+if a != 0 and b != 0:
```
Comment (screwyprof) [inline m2/main.py]: Unnecessary expression. The line 18 has already has all the checks needed

File: m2/right_triangle.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import math
+
+
+def right_triangle(a: float, b: float) -> [float, float]:
```
Comment (screwyprof) [inline m2/right_triangle.py]: Invalid function signature

File: m2/right_triangle_test.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+from m2.right_triangle import right_triangle
+
+
+@pytest.mark.parametrize("a, b, expected_c, expected_p, expected_s, expected_ga, expected_al, expected_be", [
+    (3, 4, 5, 12, 6, 90, 36.87, 53.13),  # simple right-triangle
+    (0, 1, 0, 0, 0, 0, 0.0, 0.0),  # a = 0
+    (1, 0, 0, 0, 0, 0, 0, 0),  # b = 0
+    (0, 0, 0, 0, 0, 0, 0, 0)  # a = 0, b = 0
+])
+def test_right_triangle(a: float, b: float, expected_c, expected_p, expected_s, expected_ga, expected_al, expected_be):
```
Comment (screwyprof) [inline m2/right_triangle_test.py]: Too many asserts per function

File: m2/right_triangle.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        return 0, 0, 0, 0, 0, 0
+
+    c = float(round(math.sqrt((a ** 2) + (b ** 2)),2))
+    p = float(round(((a + b + c)), 2))
+    s = float(round((((a * b) / 2)),2))
+    ga = int(90)  # because triangle is right-angled and one angle equals 90°
+    al = float(round((math.degrees(math.atan(a / b))), 2))
+    be = float(round((90 - al), 2))
+    if a != 0 and b != 0:
+        return c, p, s, ga, al, be
```
Comment (screwyprof) [inline m2/right_triangle.py]: the function returns too many arguments

File: m2/right_triangle.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    if a == 0 or b == 0:
+        return 0, 0, 0, 0, 0, 0
+
+    c = float(round(math.sqrt((a ** 2) + (b ** 2)),2))
+    p = float(round(((a + b + c)), 2))
+    s = float(round((((a * b) / 2)),2))
+    ga = int(90)  # because triangle is right-angled and one angle equals 90°
+    al = float(round((math.degrees(math.atan(a / b))), 2))
+    be = float(round((90 - al), 2))
+    if a != 0 and b != 0:
```
Comment (screwyprof) [inline m2/right_triangle.py]: Unnecessary validation. (Dobles the check from line 5)

File: m2/main.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+alpha = calc_alpha_via_tangent(a, b)
+beta = calc_beta_via_tangent(a, b)
+gamma = 180 - alpha - beta
+
+assert valid_angles(alpha, beta, gamma)
+
+print("a = " + str(a) + " см, b = " + str(b) + " см, c = "+str(c) + " см.")
+print("Угол α = " + str(alpha) + "°, угол β = " + str(beta) + "°, угол γ = " + str(gamma) + "°.")
+print("S = " + str(s) + " см, P = " + str(p) + " см^2.")
+exit_with_msg(0)
```
Comment (screwyprof) [inline m2/main.py]: Unnessesary function call. You don't need it here. The program exists automatically when reaches the end of the file

File: m2/right_triangle_test.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+@pytest.mark.parametrize("a, b, expected_beta", [
+    (3, 4, 53.13),
+    (2, 0, 0)
+])
+def test_calc_beta_via_tangent(a, b, expected_beta):
+    beta = calc_beta_via_tangent(a, b)
+    assert beta == expected_beta
+
+
+def test_valid_angles():
```
Comment (screwyprof) [inline m2/right_triangle_test.py]: You should add more tests to cover negative cases, i.e. When the angles are invalid, e.g.:
100 60 30

File: m2/right_triangle_test.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    assert valid_angles(90, 45, 45)
+
+
+def test_valid_sides():
```
Comment (screwyprof) [inline m2/right_triangle_test.py]: Some more test cases should be added. You should check invalid sides.

File: m2/right_triangle_test.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    assert valid_sides(3, 4, 5)
+
+
+def test_valid_right_triangle():
```
Comment (screwyprof) [inline m2/right_triangle_test.py]: The test doesn't cover the case when not a right triangle given.
